accounts-db/src/accounts_index_storage.rs

```diff
@@ -64,30 +64,40 @@ impl BgThreads {
     ) -> Self {
         // stop signal used for THIS batch of bg threads
         let local_exit = Arc::new(AtomicBool::default());
-        let handles = Some(
-            (0..threads)
-                .map(|idx| {
-                    // the first thread we start is special
-                    let can_advance_age = can_advance_age && idx == 0;
-                    let storage_ = Arc::clone(storage);
-                    let local_exit = local_exit.clone();
-                    let system_exit = exit.clone();
-                    let in_mem_ = in_mem.to_vec();
-
-                    // note that using rayon here causes us to exhaust # rayon threads and many tests running in parallel deadlock
-                    Builder::new()
-                        .name(format!("solIdxFlusher{idx:02}"))
-                        .spawn(move || {
-                            storage_.background(
-                                vec![local_exit, system_exit],
-                                in_mem_,
-                                can_advance_age,
-                            );
-                        })
-                        .unwrap()
-                })
-                .collect(),
-        );
+
+        let handles = if (cfg!(feature = "multithreaded")) {
```

Comment:
Can we make this a compile time feature flag instead of runtime? I believe the former is generally preferred so that we only compile one code path. I'm thinking something like this:
```rust
#[cfg(feature = "abc")]
...

#[cfg(not(feature = "abc")]
...
```

Reply:
Also, thoughts on using `#[cfg(target_os = "zkvm)]` instead of introducing a new flag?